src/cockpit/remote.py

```diff
@@ -15,21 +15,185 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+import asyncio
+import getpass
+import logging
+import os.path
+import re
+import socket
 
-from typing import Dict
+from typing import Dict, Optional, Tuple
 
-from .router import RoutingError, RoutingRule
+from ._vendor import ferny
+from .router import RoutingRule, Router
+from .peer import Peer, PeerError
+
+logger = logging.getLogger(__name__)
+
+
+class PasswordResponder(ferny.InteractionResponder):
+    PASSPHRASE_RE = re.compile(r"Enter passphrase for key '(.*)': ")
```

Comment:
Ya.  I'm painfully aware of this.  I tried hard to avoid it, but I don't think that we have a lot of other options here, unless we want to start patching upstream ssh....

This is the sort of stuff that got us in trouble ~5 years ago...

Reply:
I think running ssh(1) through `LC_MESSAGES=C.UTF-8` would go a long way actually (but that's a ferny change, I think?)